packages/eslint-plugin/src/rules/await-thenable.ts

```diff
@@ -62,6 +72,44 @@ export default createRule({
           });
         }
       },
+
+      ForOfStatement(node): void {
+        if (!node.await) {
+          return;
+        }
+
+        const type = services.getTypeAtLocation(node.right);
+        if (isTypeAnyType(type) || isTypeUnknownType(type)) {
+          return;
+        }
+
+        const asyncIteratorSymbol = tsutils.getWellKnownSymbolPropertyOfType(
+          type,
+          'asyncIterator',
+          checker,
+        );
+
+        if (asyncIteratorSymbol == null) {
+          context.report({
+            loc: getForStatementHeadLoc(context.sourceCode, node),
+            messageId: 'forAwaitOfNonThenable',
+            suggest: [
```

Comment:
Can we only offer this suggestion if the iterator result is not thenable?

Reply:
I don't think so because we'd need `getIterationTypesOfIterable()` or `getIteratedTypeOrElementType` or similar from TS none of which is exposed on the `checker`.

If we check arrays and tuples specifically, i.e. known iterables, this is silly since 
```ts
for (const promisedValue of await Promise.all(promises)) {
}
```
is a much better suggestion in those cases (since `await`ing in the loop is generally problematic for exception handling given a non-lazy array of promises)

Whereas with lazy sync-iterables of promises, which I don't know if we can detect, it may be better to do 
```ts
for (const promise of yieldPromises()) {
    const promisedValue = await promise;
}
```

I guess the only thing that's unambiguous here is that if you have an array or tuple of non-thenables, this suggestion is good. Maybe that's the only situation where we give a suggestion?